plugins/publish-plugin/src/main/kotlin/MultiplatformAppyxPublishPlugin.kt

```diff
@@ -1,12 +1,37 @@
 import org.gradle.api.Project
+import org.gradle.api.publish.PublicationContainer
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.getValue
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.registering
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
 
 internal class MultiplatformAppyxPublishPlugin : ProjectPlugin() {
 
-    companion object {
-        const val MULTIPLATFORM = "multiplatform"
+    override fun PublicationContainer.createPublications(project: Project) {
+        val mppExtension =
+            project.extensions.getByType(KotlinMultiplatformExtension::class.java)
+        val javadocJar by project.tasks.registering(Jar::class) {
+            archiveClassifier.set("javadoc")
+        }
+
+        val artifactId = project.extensions.getByType(ProjectPluginExtension::class.java).artifactId
+
+        mppExtension.targets.all {
+            mavenPublication {
+                if (artifactId.isNotEmpty()) {
+                    this.artifactId = "$artifactId-$targetName"
```

Comment:
I hope it works as expected.

Reply:
it does - check the artifacts on the image in the description